src/kontrol/solc_to_k.py

```diff
@@ -65,6 +65,60 @@ def solc_to_k(
     return _kprint.pretty_print(bin_runtime_definition, unalias=False) + '\n'
 
 
+@dataclass(frozen=True)
+class Input:
+    name: str
+    type: str
+    components: list[Input] = field(default_factory=list)
+    idx: int = 0
+
+    @staticmethod
+    def from_dict(_input: dict, i: int = 0) -> Input:
+        name = _input['name']
+        type = _input['type']
+        if _input.get('components') is not None and _input['type'] != 'tuple[]':
+            return Input(name, type, Input.recurse_comp(_input['components'], i), i)
+        else:
+            return Input(name, type, idx=i)
+
+    @staticmethod
+    def recurse_comp(components: dict, i: int = 0) -> list[Input]:
```

Comment:
Is this method public by intention?

Reply:
The name of this function (as well as `Component.recurse_component`) is a bit too general I think. Can you think of something more specific based on what these functions are supposed to perform?